### PR TITLE
Donation

### PR DESCRIPTION
**Pull request by Murmele**
@stefanknotzer @exactly-one-kas would you like to create an account at liberapay so I can add you?

you can try it out by replacing the kChangelogUrl in the Updater.cpp by https://raw.githubusercontent.com/Murmele/Gittyup/890306d9560008a12a54e881370ea5c5f0b12401/doc/changelog.md

- [x] ~~check installer script for linux!~~ Currently no linux installer exists, only the flatpak package and packages in the package managers
- [x] Test if setting tag works as intended

### CODE REVIEW

File: src/dialogs/AboutDialog.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 namespace {
 
-const QString kEmail = QStringLiteral("TODO");
+const QString kEmail = QStringLiteral("https://github.com/Murmele/Gittyup/issues");
```
Comment (exactly-one-kas) [inline src/dialogs/AboutDialog.cpp:33]: - [ ] Rename this variable to something like `kIssueTracker`

File: rsrc/Liberapay_Icon_Readme.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ source: https://icon-icons.com/icon/liberapay/130890
```
Comment (exactly-one-kas) [inline rsrc/Liberapay_Icon_Readme.md:1]: - [ ] I think the attribution also needs to be visible from the compiled App

File: src/ui/MenuBar.cpp [hunk 3]
```diff
@@ -1035,6 +1037,11 @@ MenuBar::MenuBar(QWidget *parent)
     QDesktopServices::openUrl(QUrl::fromLocalFile(url));
   });
 
+  QAction* donation = help->addAction(tr("Support us over Liberapay"));
```
Comment (exactly-one-kas) [inline src/ui/MenuBar.cpp:1040]: - [ ] I think this should read "Support us *via* Liberapay"

File: src/update/Updater.cpp [hunk 4]
```diff
@@ -75,24 +75,23 @@ Updater::Updater(QObject *parent)
 
   connect(this, &Updater::updateAvailable,
   [this](const QString &version, const QString &log, const QString &link) {
```
Comment (exactly-one-kas) [inline src/update/Updater.cpp:77]: - [ ] Resolve the TODO in line 47

**Review by exactly-one-kas — CHANGES_REQUESTED** [review]

**Comment by exactly-one-kas** [comment]
@Murmele Thanks for the offer, but I already have a steady source of income

File: rsrc/Liberapay_Icon_Readme.md (hunk 2 above)
Comment (Murmele) [reply rsrc/Liberapay_Icon_Readme.md]: I added the source to the Acknowledments

**Comment by Murmele** [comment]
@exactly-one-kas I added a messagebox if the platform is linux (and not flatpak) if it should be really downloaded manually. GitAhead had a shell script which extracts it self. I don't like that people are installing it without package manager because it is hard to remove everything at uninstall

**Comment by Murmele** [comment]
I added to every step the following condition:
`if: ${{ github.ref == 'refs/heads/master' || env.IS_RELEASE}}`
The problem is that for some reason the env is not available in the jobs. Maybe you know something about?

**Comment by exactly-one-kas** [comment]
> I added a messagebox [...]

Manual installation and updating shouldn't be too much of a problem if it's inside a self-contained directory, especially if you want to use bleeding-edge builds.

Also, wouldn't it be better to simply remove the self-update button for linux binary builds?

> The problem is that for some reason the env is not available in the jobs

The environment variable is available, but it's a string, so it evaluates to `true` (only empty strings are considered falsey).
This should work using `${{ github.ref == 'refs/heads/master' || env.IS_RELEASE != 'false' }}`

**Comment by Murmele** [comment]
> Also, wouldn't it be better to simply remove the self-update button for linux binary builds?

Yeah maybe thats a good idea. So I remove any update functionality on linux? Maybe I just write down that system should be updated to receive the newest version?

**Comment by Murmele** [comment]
> 

when I tried to add it only for the job I got the following error message:
`The workflow is not valid. .github/workflows/build.yml (Line: 270, Col: 9): Unrecognized named-value: 'env'. Located at position 1 within expression: env.IS_RELEASE`
https://github.com/Murmele/Gittyup/actions/runs/2212832925

**Comment by exactly-one-kas** [comment]
Yes, the `if` on the whole job doesn't work with env (since there's no environment without a started VM / container)
You can only use the `if`s on steps.

**Comment by Murmele** [comment]
Flatpak package:
image

**Comment by Murmele** [comment]
> Yes, the `if` on the whole job doesn't work with env (since there's no environment without a started VM / container) You can only use the `if`s on steps.

I just copied now the IS_RELEASE condition again into the job if, instead having in every step the condition

File: src/update/UpdateDialog.cpp [hunk 5]
```diff
@@ -52,54 +56,94 @@ UpdateDialog::UpdateDialog(
...
-  content->addWidget(download);
+  content->addLayout(l);
   content->addWidget(buttons);
 
   QHBoxLayout *layout = new QHBoxLayout(this);
   layout->addLayout(iconLayout);
   layout->addLayout(content);
 
-  connect(this, &UpdateDialog::accepted, [link] {
+  connect(this, &UpdateDialog::accepted, [/*this,*/ platform, link] {
```
Comment (exactly-one-kas) [inline src/update/UpdateDialog.cpp:144]: Please remove the comment

File: src/update/Updater.cpp [hunk 6]
```diff
@@ -44,7 +44,7 @@ namespace {
 
 const QString kTemplateFmt = "%1-XXXXXX.%2";
 const QString kLinkFmt =
-  "TODO: replace by release link<https://github.com/gitahead/gitahead/releases/download/v%1/GitAhead%2-%3.%4>";
+  "link<https://github.com/Murmele/gittyup/releases/download/v%1/Gittyup%2-%3.%4>";
```
Comment (exactly-one-kas) [inline src/update/Updater.cpp:47]: This should only have the URL part

**Review by exactly-one-kas — CHANGES_REQUESTED** [review]

File: src/update/UpdateDialog.cpp (hunk 5 above)
Comment (Murmele) [reply src/update/UpdateDialog.cpp:144]: done

**Pull request merged**